### PR TITLE
CRAFTERCMS-1725

### PR DESCRIPTION
**Pull request by dejan-brkic**
Added studio integration with remote repos

### Ticket reference or full description of what's in the PR
https://github.com/craftercms/craftercms/issues/1725

### CODE REVIEW

File: src/main/java/org/craftercms/studio/impl/v1/repository/git/GitContentRepository.java [hunk 1]
```diff
@@ -1398,6 +1404,50 @@ public void markGitLogVerifiedProcessed(String siteId, String commitId) {
         gitLogMapper.markGitLogProcessed(params);
     }
 
+    @Override
+    public boolean createSiteCloneRemote(String siteId, String remoteName, String remoteUrl, String remoteUsername, String remotePassword) {
+        boolean toReturn;
+
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v1/repository/git/GitContentRepository.java:1408]: Please add debug logs to happy path.

File: src/main/java/org/craftercms/studio/impl/v1/repository/git/GitContentRepository.java [hunk 2]
```diff
@@ -1398,6 +1404,50 @@ public void markGitLogVerifiedProcessed(String siteId, String commitId) {
...
+        // clone remote git repository for site content
+        toReturn = helper.createSiteCloneRemoteGitRepo(siteId, remoteName, remoteUrl, remoteUsername, remotePassword);
+
+        if (toReturn) {
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v1/repository/git/GitContentRepository.java]: you need nested ifs to locate errors and report them

File: src/main/java/org/craftercms/studio/impl/v1/service/site/SiteServiceImpl.java [hunk 3]
```diff
@@ -569,7 +568,298 @@ private void addDefaultGroupsForNewSite(String siteId) {
...
+        switch (createOption) {
+            case REMOTE_REPOSITORY_CREATE_OPTION_CLONE:
+                createSiteCloneRemote(siteId, description, remoteName, remoteUrl, remoteUsername, remotePassword);
+                break;
+
+            case REMOTE_REPOSITORY_CREATE_OPTION_PUSH:
+                createSitePushToRemote(siteId, description, blueprintName, remoteName, remoteUrl, remoteUsername, remotePassword);
+                break;
+
+            default:
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v1/service/site/SiteServiceImpl.java:586]: Log error/invalid state?

**Review by sumerjabri — CHANGES_REQUESTED** [review]

**Comment by dejan-brkic** [comment]
Added requested changes

**Pull request merged**